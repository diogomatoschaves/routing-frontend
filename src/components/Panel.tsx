import React from 'react'
import styled from 'styled-components'
import { Box } from '../styledComponents'
import {
  GeographiesHandler,
  HandleConfirmButton,
  HandleDeleteRoute,
  HandleValueUpdate,
  InputColors,
  InputValues,
  LocationInfo,
  OptionsHandler,
  ProfileItem,
  Route,
  UpdatePoint,
  UpdateState
} from '../types'
import DebugPanel from './DebugPanel'
import DefaultPanel from './DefaultPanel'
import Tabs from './Tabs'

interface Props {
  updatePoint: UpdatePoint
  handleValueUpdate: HandleValueUpdate
  handleAddRoute: HandleConfirmButton
  handleClickRoute: HandleConfirmButton
  handleDeleteRoute: HandleDeleteRoute
  handleShowClick: () => void
  locations: LocationInfo[]
  profiles: ProfileItem[]
  routingGraphVisible: boolean
  polygonsVisible: boolean
  googleMapsOption: boolean
  trafficOption: boolean
  updateState: UpdateState
  geographies: GeographiesHandler
  profile: string
  duration: number
  urlMatchString: string
  debug: boolean
  modeTabsHandler: OptionsHandler
  addDataTabsHandler: OptionsHandler
  inputValues: InputValues
  inputColors: InputColors
  addedRoutes: Route[]
  loading: boolean
}

const PanelWrapper: any = styled(Box)`
  position: absolute;
  z-index: 1000;
  width: 30%;
  min-width: 350px;
  max-width: 450px;
  max-height: 80vh;
  left: 40px;
  top: 40px;
  padding: 25px 25px 35px 25px;
  background: rgba(255, 255, 255, 0.92);
  border-radius: 10px;
  box-shadow: 10px 10px 16px -9px rgba(77, 77, 77, 0.5);
  transition: flex 1s ease-out;
  overflow: auto;
`

const Panel: any = (props: Props) => {
  const {
    updatePoint,
    handleAddRoute,
    handleDeleteRoute,
    handleClickRoute,
    locations,
    profiles,
    routingGraphVisible,
    polygonsVisible,
    googleMapsOption,
    updateState,
    geographies,
    profile,
    urlMatchString,
    trafficOption,
    debug,
    modeTabsHandler,
    handleValueUpdate,
    inputValues,
    inputColors,
    addDataTabsHandler,
    addedRoutes,
    loading,
  } = props

  return (
    <PanelWrapper
      as={PanelWrapper}
      direction="column"
      justify="flex-start"
      animation={'scale'}
      duration={500}
      flex={debug}
    >
      <Box direction="row" justify="flex-start" padding="0 0 10px 0">
        <Tabs
          tabsHandler={modeTabsHandler}
          updateState={updateState}
          id="modeTabsHandler"
          width="100px"
          justify="flex-start"
        />
      </Box>
      {debug ? (
        <DebugPanel
          routingGraphVisible={routingGraphVisible}
          polygonsVisible={polygonsVisible}
          updateState={updateState}
          geographies={geographies}
          handleValueUpdate={handleValueUpdate}
          handleAddRoute={handleAddRoute}
          handleDeleteRoute={handleDeleteRoute}
          handleClickRoute={handleClickRoute}
          inputValues={inputValues}
          inputColors={inputColors}
          addDataTabsHandler={addDataTabsHandler}
          addedRoutes={addedRoutes}
        />
      ) : (
        <DefaultPanel
          updatePoint={updatePoint}
          locations={locations}
          routingGraphVisible={routingGraphVisible}
          polygonsVisible={polygonsVisible}
          googleMapsOption={googleMapsOption}
          updateState={updateState}
          geographies={geographies}
          profile={profile}
          urlMatchString={urlMatchString}
          trafficOption={trafficOption}
          profiles={profiles}
          loading={loading}
        />
      )}
    </PanelWrapper>
  )
}

export default Panel
