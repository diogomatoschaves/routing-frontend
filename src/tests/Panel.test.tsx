import React from 'react'
import { MemoryRouter } from 'react-router-dom'
import TestRenderer from 'react-test-renderer'
import Panel from '../components/Panel'

const mockLocations = [
  {
    lat: null,
    lng: null,
    marker: 'map marker alternate',
    name: 'start',
    placeholder: 'Origin'
  },
  {
    lat: null,
    lng: null,
    marker: 'flag checkered',
    name: 'end',
    placeholder: 'Destination'
  }
]

const mockTabsHandler = {
  activeIdx: 0,
  options: [
    { key: 'default', text: 'Interactive', value: 0 },
    { key: 'debug', text: 'Debugging', value: 1 }
  ]
}

const mockGeographies = [
  {
    coords: [13.38408, 52.51721],
    name: 'Berlin',
    polygon: 'berlin.geojson'
  },
  {
    coords: [9.033, 48.7111],
    name: 'Stuttgart',
    polygon: 'stuttgart.geojson'
  },
  {
    coords: [8.7214, 47.912],
    name: 'Immendingen',
    polygon: 'immendingen.geojson'
  },
  {
    coords: [-121.97588, 37.34606],
    name: 'San JosÃ©',
    polygon: 'sunnyvale.geojson'
  }
]

const profiles = [
  {
    iconName: 'car',
    name: 'car'
  },
  {
    iconName: 'male',
    name: 'foot'
  },
  {
    iconName: 'rocket',
    name: 'pilot'
  }
]

it('expect to render Panel component', () => {
  const testInstance = TestRenderer.create(
    <MemoryRouter initialEntries={['/']}>
      <Panel
        locations={mockLocations}
        profiles={profiles}
        updatePoint={jest.fn()}
        updateState={jest.fn()}
        handleValueUpdate={jest.fn()}
        handleShowClick={jest.fn()}
        routingGraphVisible={false}
        polygonsVisible={false}
        googleMapsOption={false}
        recenter={false}
        authorization={''}
        geographies={mockGeographies}
        geography={mockGeographies[0]}
        debug={false}
        modeTabsHandler={mockTabsHandler}
      />
    </MemoryRouter>
  )
  expect(testInstance.toJSON()).toMatchSnapshot()
})
