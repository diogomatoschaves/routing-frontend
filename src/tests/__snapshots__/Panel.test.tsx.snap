// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render Panel component 1`] = `
<div
  className="sc-bdVaJa sc-chPdSV VEVzz"
  direction="column"
>
  <div
    className="scale visible transition sc-bdVaJa pqpqB"
    direction="row"
    style={
      Object {
        "animationDuration": undefined,
      }
    }
  >
    <div
      className="sc-bdVaJa kRVJAa"
      direction="row"
    >
      <div
        className="sc-bdVaJa sc-fjdhpX TpKDF"
        color={true}
        id="default"
        onClick={[Function]}
        width="100px"
      >
        Interactive
      </div>
      <div
        className="sc-bdVaJa sc-fjdhpX jAZwpU"
        color={false}
        id="debug"
        onClick={[Function]}
        width="100px"
      >
        Debugging
      </div>
    </div>
  </div>
  <div
    className="sc-bdVaJa jsucjq"
    direction="column"
  >
    <div
      className="sc-bdVaJa fuiAsE"
      direction="row"
    >
      <div
        id="car-profile"
        onClick={[Function]}
      >
        <div
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <div
            className="sc-bwzfXH deebee"
            color="transparent"
            cursor="pointer"
          >
            <div
              className="sc-bdVaJa dbHwHO"
              height="100%"
            >
              <i
                aria-hidden="true"
                className="car icon sc-htpNat bxTmCt"
                fontSize="25px"
                height="20px"
                onClick={[Function]}
                overridecolor="#a6cad8"
                padding="0 0 0 0"
                position="absolute"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        id="foot-profile"
        onClick={[Function]}
      >
        <div
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <div
            className="sc-bwzfXH deebee"
            color="transparent"
            cursor="pointer"
          >
            <div
              className="sc-bdVaJa dbHwHO"
              height="100%"
            >
              <i
                aria-hidden="true"
                className="male icon sc-htpNat bxTmCt"
                fontSize="25px"
                height="20px"
                onClick={[Function]}
                overridecolor="#a6cad8"
                padding="0 0 0 0"
                position="absolute"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="sc-bdVaJa gSlpgJ"
      direction="row"
    >
      <div
        className="sc-bwzfXH jWeZli"
        color="#007a93"
      >
        <div
          className="sc-bdVaJa lajepV"
          height="100%"
        >
          <i
            aria-hidden="true"
            className="map marker alternate icon sc-htpNat iIqnlv"
            fontSize="25px"
            height="20px"
            onClick={[Function]}
            overridecolor="white"
            padding="0 0 0 0"
            position="absolute"
          />
        </div>
      </div>
      <div
        className="ui fluid input sc-dnqmqq bSXmIE"
      >
        <input
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="Origin"
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      className="sc-bdVaJa gSlpgJ"
      direction="row"
    >
      <div
        className="sc-bwzfXH jWeZli"
        color="#007a93"
      >
        <div
          className="sc-bdVaJa lajepV"
          height="100%"
        >
          <i
            aria-hidden="true"
            className="flag checkered icon sc-htpNat iIqnlv"
            fontSize="25px"
            height="20px"
            onClick={[Function]}
            overridecolor="white"
            padding="0 0 0 0"
            position="absolute"
          />
        </div>
      </div>
      <div
        className="ui fluid input sc-dnqmqq bSXmIE"
      >
        <input
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="Destination"
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      className="sc-bdVaJa gSlpgJ"
      direction="row"
    >
      <div
        className="sc-bdVaJa dbyCSG"
        direction="row"
        height="50px"
        width="65%"
      >
        <div
          className="ui fitted toggle checkbox custom-class sc-gZMcBi hupRmQ"
          onChange={[Function]}
          onClick={[Function]}
          onMouseDown={[Function]}
          onMouseUp={[Function]}
        >
          <input
            checked={false}
            className="hidden"
            id="googleMapsOption"
            readOnly={true}
            tabIndex={0}
            type="checkbox"
          />
          <label
            htmlFor="googleMapsOption"
          />
        </div>
        <label
          className="sc-iwsKbI lpcEmi"
        >
          Google Maps
        </label>
      </div>
      <div
        className="sc-bdVaJa gpVhCu"
        direction="row"
        height="50px"
        width="35%"
      >
        <div
          className="ui fitted toggle checkbox custom-class sc-gZMcBi hupRmQ"
          onChange={[Function]}
          onClick={[Function]}
          onMouseDown={[Function]}
          onMouseUp={[Function]}
        >
          <input
            checked={false}
            className="hidden"
            id="trafficOption"
            readOnly={true}
            tabIndex={0}
            type="checkbox"
          />
          <label
            htmlFor="trafficOption"
          />
        </div>
        <label
          className="sc-iwsKbI lpcEmi"
        >
          Traffic
        </label>
      </div>
    </div>
  </div>
</div>
`;
